proposals/0000-arkworks-integration.md

```diff
@@ -0,0 +1,95 @@
+---
+Title: Arkworks Integration
+Number: 0
+Status: Proposed
+Authors:
+  - Achim Schneider<achim@parity.io>
+Created: 2023-13-03
+Category: Runtime Environment
+Requires:
+Replaces:
+--- 
+
+## Summary
+ Proposal to add host functions for [Arkworks](http://arkworks.rs/) to Polkadot. We provide Elliptic curves (bls12_381, bls12_377, ed_on_bls12_381, ed_on_bls12_77, bw6_761), were we replace the compute intense operations by host function calls. We add those host function calls to `sp_io::elliptic_curves` and refer to the `arkworks` crate. 
+The curves live in https://github.com/paritytech/ark-substrate and call into the host functions. To avoid point preparation of the elliptic curves in the runtime, we also partially fork the models `bls12` and `bw6` in [ark-substrate](https://github.com/paritytech/ark-substrate).
+
+## Motivation
+Usually cryptographic operations on elliptic curves are slow in WebAssembly. Replacing those operations by host functoin calls into binary code allows us to drastically improve the performace of the cryptographic tools which rely on those operations.
+
+## Detailed Solution design
+We implement host function calls into the underlying arithmetic operations on elliptic curves for `BLS12_377`, `BLS12_381`, `BW6_761`, `ED_ON_BLS12_377` and `ED_ON_BLS12_381` by host function calls.
```

Comment:
Should mention the host functions that are being added. Like it is done here: https://github.com/w3f/PPPs/pull/7

Reply:
Done